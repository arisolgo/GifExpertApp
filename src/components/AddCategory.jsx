import { useState } from "react";
export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");
  const handleInputChange = ({ target }) => {
    setInputValue(target.value);
    console.log(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;
    onNewCategory(inputValue);
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Search for Gifs"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  );
};
