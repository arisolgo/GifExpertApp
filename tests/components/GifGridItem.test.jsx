import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components/GifGridItem";

describe("Tests in <GifGridItem/>", () => {
  const title = "SetImagesTitle";
  const url = "https://selectedtopic.com/selectedtopic.jpg";

  test("should match with the snapshot", () => {
    const { container } = render(<GifGridItem title="fdf" url="dfd" />);
    expect(container).toMatchSnapshot();
  });

  test("should have the given url as src and title as alt", () => {
    render(<GifGridItem title={title} url={url} />);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(title).toBe(alt);
  });

  test("should show the title in the component", () => {
    render(<GifGridItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
