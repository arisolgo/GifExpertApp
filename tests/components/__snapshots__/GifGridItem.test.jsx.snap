// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGridItem/> should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="fdf"
      src="dfd"
    />
    <p>
      fdf
    </p>
  </div>
</div>
`;
